constph.py

```diff
@@ -588,28 +633,116 @@ def setTitrationState(self, titration_group_index, titration_state_index, contex
             # TODO: Handle Custom forces.
             if force_classname == 'NonbondedForce':
                 for exception_index in titration_group['exception_indices']:
-                    [particle1, particle2, chargeProd, sigma, epsilon] = force.getExceptionParameters(exception_index)
-                    [charge1, sigma1, epsilon1] = force.getParticleParameters(particle1)
-                    [charge2, sigma2, epsilon2] = force.getParticleParameters(particle2)
-                    #print "chargeProd: old %s new %s" % (str(chargeProd), str(self.coulomb14scale * charge1 * charge2))
+                    [particle1, particle2, chargeProd, sigma, epsilon] = map(strip_in_unit_system,
```

Comment:
We want to avoid all of these `get*` calls, since this incurs overhead, by caching all of the parameters we need for `force.setExceptionParameters(...)`.